### ISSUE TITLE
Clear Audio Device hang if Stop Audio Device called just prior in parallel

### ISSUE THREAD

**Issue by dataflowg** [dataflowg/g-audio#20]
This issue is reproducible by running and stopping `Audio Loopback Example.vi` multiple times. After anywhere from 1-10 runs `Clear Audio Device.vi` appears to hang and never returns, and on closing the project there's the dreaded Resetting VI message from LabVIEW.

Running the debug build shows `ma_worker_thread()` is throwing an assert failure:
``` c
/* We wait on an event to know when something has requested that the device be started and the main loop entered. */
ma_event_wait(&pDevice->wakeupEvent);

/* Default result code. */
pDevice->workResult = MA_SUCCESS;

/* If the reason for the wake up is that we are terminating, just break from the loop. */
if (ma_device_get_state(pDevice) == ma_device_state_uninitialized) {
    break;
}

/*
Getting to this point means the device is wanting to get started. The function that has requested that the device
be started will be waiting on an event (pDevice->startEvent) which means we need to make sure we signal the event
in both the success and error case. It's important that the state of the device is set _before_ signaling the event.
*/
MA_ASSERT(ma_device_get_state(pDevice) == ma_device_state_starting);
```
where the value of `ma_device_get_state(pDevice)` is `ma_device_state_stopped`. It should instead be `ma_device_state_uninitialized`, and handled by the earlier guard condition. I suspect a race condition in stopping and clearing the audio device from different threads in LabVIEW is causing `ma_event_wait()` to return before the correct value for the device state has been set.

One possible solution is to ensure `stop_audio_device()` waits until the device has actually stopped using `ma_event_wait()`, or polling `ma_device_get_state()`.

**Comment by dataflowg** [comment]
Fixed in commit [a3fa91a](https://github.com/dataflowg/g-audio/commit/a3fa91a640592b65c975faa99d1b5c6a5a562418). Solution uses a device level mutex when stopping and clearing the device.

**Comment by dnattinger** [comment]
@dataflowg I was able to reproduce this hang in LabVIEW 2024 Q3 Patch 2 64-bit and G-Audio 0.4.0.1, which I'm guessing doesn't include this fix, judging by the release date of the package vs. the date of your commit above. Were you planning on releasing a new G-Audio package on vipm.io with this fix at some point? Or am I going to have to launch Visual Studio for the first time in my life to try to build the updated DLL? (please say no)

Alternatively, I've been testing my code without calling _Clear Audio Backend.vi_ and it seems to be working fine. I read the docs/comments in the code and I'm still not sure why I need to call this VI (or not call it)?
